### ISSUE TITLE
Train name keeps getting added "Split off from:" when decoupling leading to a fatal crash when the string overflows

### ISSUE THREAD

**Issue by d-513** [Layers-of-Railways/Railway#406]
### Describe the Bug

Train name keeps getting added "Split off from:" when decoupling leading to a fatal crash when the string overflows the max character limit

This has also the effect of not being able to read the train coupling block's tooltip as the train's name doesn't fit on the screen.

### Reproduction Steps

1. Create a train
2. Make any kind of automated system that will decouple cars using the coupler blocks (e.g sending a redstone link signal via train schedule to a redstone link next to a decoupling block). Get a chunk loader so the signal works even when no one is playing.
3. Every time it's decoupled and recoupled, a "Split off from: " is added to the front of the train's name.
4. Keep it running for a couple of days, server will crash due to overflow

### Expected Result

"Split off from:" being removed when recoupling, or using a randomly generated string to identify the split off's
A config option to disable the renaming would be fine as well.

### Screenshots and Videos

![Screenshot_20240321_102449](https://github.com/Layers-of-Railways/Railway/assets/49000790/222cd0ee-f075-4e20-b589-3042df7a882a)
My station coupling setup:

![Screenshot_20240321_102438](https://github.com/Layers-of-Railways/Railway/assets/49000790/39c2156d-8be6-4676-9d40-ca05b068b880)

### Crash Report or Log

https://mclo.gs/OhinZt3

### Operating System

Debian 12

### Mod Version

1.5.3

### Create Mod Version

0.5.1f

### Minecraft Version

1.20.1

### ModLoader and Version

Forge

### Other Mods

![image](https://github.com/Layers-of-Railways/Railway/assets/49000790/6235c48a-9313-48b8-baaf-0b312491e4b2)

### Additional Context

It attempts that an attempt was made to remedy this issue, however it proved insufficient 
```
train.doubleEnded = train.carriages.stream().anyMatch(carriage -> carriage.anyAvailableEntity().getContraption() i>
        if(!train.name.getString().contains("Split off from: ")){
            newTrain.name = Components.literal("Split off from: "+train.name.getString());
        }
        else newTrain.name = Components.literal(train.name.getString());
```
This condition should prevent this from happening, but it does not. There must be an issue here or somewhere else in the code.

**FIX FOR WORLD BROKEN BY THIS**
Warning: probably will not give you back the shedule, have a backup
1. Download world/data/create_tracks.dat
2. Open nbt editor e.g https://irath96.github.io/webNBT/
3. Edit the train name to something that doesn't overflow

**Comment by IThundxr** [comment]
This should be fixed by https://github.com/Layers-of-Railways/Railway/commit/48d26db12843f9b0fe2ba8390f512cc4a795bbe5 already

**Comment by d-513** [comment]
> This should be fixed by [48d26db](https://github.com/Layers-of-Railways/Railway/commit/48d26db12843f9b0fe2ba8390f512cc4a795bbe5) already


Yes, I saw the code, but the issue hasn’t been fixed. I have steam and rails 1.5.3

edit: unless that hasn’t been in the release yet

**Comment by IThundxr** [comment]
> > This should be fixed by [48d26db](https://github.com/Layers-of-Railways/Railway/commit/48d26db12843f9b0fe2ba8390f512cc4a795bbe5) already
> 
> 
> 
> 
> 
> Yes, I saw the code, but the issue hasn’t been fixed. I have steam and rails 1.5.3
> 
> 
> 
> edit: unless that hasn’t been in the release yet

Correct, it's scheduled to be released in 1.6.0
